File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.go
func longestSubsequence(nums []int) int {
    n := len(nums)
    xor := 0
    hasNonZero := false

    for _, num := range nums {
        xor ^= num
        if num != 0 {
            hasNonZero = true
        }
    }

    if xor != 0 {
        return n
    }
    if hasNonZero {
        return n - 1
    }
    return 0
}

File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& nums) {
        int n = nums.size();
        int xorVal = 0;
        bool hasNonZero = false;

        for (int num : nums) {
            xorVal ^= num;
            if (num != 0) {
                hasNonZero = true;
            }
        }

        if (xorVal != 0) {
            return n;
        }
        return hasNonZero ? n - 1 : 0;
    }
};
